ueberdosis/tiptap
# Fix React Dependency Array creating new editor instances instead of reusing existing one

## Please describe your changes

Before this PR everytime dependencies change the whole editor gets replaced with a new instance. This would remove entered content, cause flashes and gets rid of the editor state (which sucks for example because of the history living in the state for example)

This PR implements a check that will update the editor options when the deps change instead of creating a new instance with said options.

## How did you accomplish your changes

Added a check in the react useEditor hook to update the editor instead of replacing it.

## How have you tested your changes

I tested my changes on the demo page I created (Link: https://deploy-preview-4453--tiptap-embed.netlify.app/src/issues/2403/react/)

## How can we verify your changes

See above

1. Write something - in the console you should see `0`
2. Click on Inc a few times
3. Write something - now the onUpdate function inside the editor should be updated
4. Check the editor class (it should be `my-editor`
5. Click on the Update class button
6. The editor class should now correctly be `my-editor updated-editor`

## Remarks

On my tests everything worked fine for now but I think we'd need to verify that this change is not breaking some special cases.

## Checklist


## Related issues

- fixes #2403
